### PR TITLE
feat: 添加插件详情页与搜索支持

### PR DESCRIPTION
**Pull request by eya46**
#23 
实现 `https://registry.nonebot.dev/search/?q=nonebot`

### CODE REVIEW

**Comment by eya46** [comment]
git炸完了，一直在搞git，我觉得我要寄了

**Comment by he0119** [comment]
> git炸完了，一直在搞git，我觉得我要寄了

你要不直接 reset --hard 然后重新 push -f？

File: src/router/Home.vue [hunk 1]
```diff
@@ -11,7 +11,6 @@ import { usePageStore } from "@/stores/page";
 const route = useRoute();
 const store = usePageStore();
 const { plugins, results } = storeToRefs(store);
-store.initData();
```
Comment (he0119) [inline src/router/Home.vue]: 说起来这个 initData 不应该就放在 home 里面吗？如果我进入 404 界面是不需要加载数据的（

Comment (eya46) [reply src/router/Home.vue]: 好像也是（

**Comment by eya46** [comment]
[nonebot-plugin-status:nonebot_plugin_status](https://deploy-preview-24--nonebot-registry.netlify.app/plugin/nonebot-plugin-status:nonebot_plugin_status)
plugin页面做啥样的

还有这个怎么改
想改成不够高时不占满整个屏幕，然后footer固定在底部...改不出来
```html
<n-layout class="my-3 min-h-screen">
  <router-view></router-view>
</n-layout>
```

**Comment by he0119** [comment]
> [nonebot-plugin-status:nonebot_plugin_status](https://deploy-preview-24--nonebot-registry.netlify.app/plugin/nonebot-plugin-status:nonebot_plugin_status) plugin页面做啥样的
> 
> 还有这个怎么改 想改成不够高时不占满整个屏幕，然后footer固定在底部...改不出来
> 
> ```
> <n-layout class="my-3 min-h-screen">
>   <router-view></router-view>
> </n-layout>
> ```

感觉展示一下基本信息，再加上测试情况就可以吧。比如当前测试时间，测试版本，测试结果。

**Comment by eya46** [comment]
问题：
- plugin.tags.length
- result.outputs.validation
- ResultTable里面的类型
```vue
PluginPage 110行、178行
<n-button v-if="plugin.tags.length === 0" text>无</n-button>

<n-card title="验证结果" size="huge" hoverable>
  <pre class="max-h-[50vh] overflow-auto">{{
    result.outputs.validation
  }}</pre>
</n-card>
```

```
# result.outputs.validation 也是差不多的报错
Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'length')
    at PluginPage.vue:110:37
    at renderFnWithContext (runtime-core.esm-bundler.js:766:13)
    at resolveWrappedSlot (resolve-slot.js:32:47)
    at Proxy.render (Card.js:130:13)
    at renderComponentRoot (runtime-core.esm-bundler.js:816:16)
    at ReactiveEffect.componentUpdateFn [as fn] (runtime-core.esm-bundler.js:5701:46)
    at ReactiveEffect.run (reactivity.esm-bundler.js:178:19)
    at instance.update (runtime-core.esm-bundler.js:5814:51)
    at setupRenderEffect (runtime-core.esm-bundler.js:5822:5)
    at mountComponent (runtime-core.esm-bundler.js:5612:5)
```

**Comment by eya46** [comment]
> 问题：
> 
> * plugin.tags.length
> * result.outputs.validation
> * ResultTable里面的类型
> 
> ```
> PluginPage 110行、178行
> <n-button v-if="plugin.tags.length === 0" text>无</n-button>
> 
> <n-card title="验证结果" size="huge" hoverable>
>   <pre class="max-h-[50vh] overflow-auto">{{
>     result.outputs.validation
>   }}</pre>
> </n-card>
> ```
> 
> ```
> # result.outputs.validation 也是差不多的报错
> Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'length')
>     at PluginPage.vue:110:37
>     at renderFnWithContext (runtime-core.esm-bundler.js:766:13)
>     at resolveWrappedSlot (resolve-slot.js:32:47)
>     at Proxy.render (Card.js:130:13)
>     at renderComponentRoot (runtime-core.esm-bundler.js:816:16)
>     at ReactiveEffect.componentUpdateFn [as fn] (runtime-core.esm-bundler.js:5701:46)
>     at ReactiveEffect.run (reactivity.esm-bundler.js:178:19)
>     at instance.update (runtime-core.esm-bundler.js:5814:51)
>     at setupRenderEffect (runtime-core.esm-bundler.js:5822:5)
>     at mountComponent (runtime-core.esm-bundler.js:5612:5)
> ```

还有个要解决的，plugins和results缓存，写了半天没法用

**Comment by he0119** [comment]
> > 问题：
> > 
> > * plugin.tags.length
> > * result.outputs.validation
> > * ResultTable里面的类型
> > 
> > ```
> > PluginPage 110行、178行
> > <n-button v-if="plugin.tags.length === 0" text>无</n-button>
> > 
> > <n-card title="验证结果" size="huge" hoverable>
> >   <pre class="max-h-[50vh] overflow-auto">{{
> >     result.outputs.validation
> >   }}</pre>
> > </n-card>
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > ```
> > # result.outputs.validation 也是差不多的报错
> > Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'length')
> >     at PluginPage.vue:110:37
> >     at renderFnWithContext (runtime-core.esm-bundler.js:766:13)
> >     at resolveWrappedSlot (resolve-slot.js:32:47)
> >     at Proxy.render (Card.js:130:13)
> >     at renderComponentRoot (runtime-core.esm-bundler.js:816:16)
> >     at ReactiveEffect.componentUpdateFn [as fn] (runtime-core.esm-bundler.js:5701:46)
> >     at ReactiveEffect.run (reactivity.esm-bundler.js:178:19)
> >     at instance.update (runtime-core.esm-bundler.js:5814:51)
> >     at setupRenderEffect (runtime-core.esm-bundler.js:5822:5)
> >     at mountComponent (runtime-core.esm-bundler.js:5612:5)
> > ```
> 
> 还有个要解决的，plugins和results缓存，写了半天没法用

好的，晚上我来看看。

**Pull request merged**